source/nodejs/adaptivecards-designer/src/open-image-dialog.ts

```diff
@@ -0,0 +1,324 @@
+        title.innerText = "Upload your own card design";
+        return title;
+    }
+
+    private renderTitleContent():HTMLElement {
+        const content = document.createElement("div");
+        content.className="acd-image-title-content";
+        content.innerText = "Design your cards and upload it as a png, or jpeg and see it magically convert to a functioning adaptive card";
+        return content;
+    }
+
+    private renderUploadedImage():HTMLElement {
+        this._imageElement = document.createElement("img");
+        this._imageElement.className="acd-image-placeholder";
+        this._imageElement.setAttribute('src', this._uploadedImage);
+        return this._imageElement;
+    }
+
+    private renderUploadButton(type: string):HTMLElement {
+        this._buttonContainer = document.createElement("div");
+        this._buttonContainer.className="acd-upload-button-container";
+        const fileNote = document.createElement("div");
+        fileNote.className="acd-image-file-note"
+        fileNote.innerHTML= "By using the upload feature you agree to our <a href='../privacy' target='_blank'><span class='link'> image usage policy</spanl></a>";
+        type !== 'file' && this._buttonContainer.appendChild(this.renderCheckbox());
+        this._buttonContainer.appendChild(type === 'file' ? this.createFileButton(): this.createActionButton());
+        this._buttonContainer.appendChild(fileNote);
+        return this._buttonContainer;
+    }
+    
+    private renderCheckbox() {
+        const checkboxContainer = document.createElement("div");
+        checkboxContainer.className="acd-checkbox-container"
+        const inputElement = <HTMLInputElement>document.createElement("INPUT");       
+        inputElement.className="acd-checkbox-element";
+        inputElement.setAttribute("id","template-option");
+        inputElement.setAttribute("type", "checkbox");
+        inputElement.onclick = () => {
+            const checkbox = document.getElementById("template-option") as HTMLInputElement;
+            this.isTemplateOptionChecked = checkbox.checked ? true: false
+        }
+        const customCheckboxElement = <HTMLInputElement>document.createElement("SPAN");
+        customCheckboxElement.className="acd-custom-checkbox";
+        checkboxContainer.appendChild(inputElement);
+        checkboxContainer.appendChild(customCheckboxElement);
+        checkboxContainer.appendChild(this.renderLabelText());
+        return checkboxContainer;
+    }
+
+    private renderLabelText()  {
+        const labelElement = <HTMLInputElement>document.createElement("LABEL");       
+        labelElement.className = "acd-label-element"
+        labelElement.setAttribute("for", "template-option");
+        labelElement.innerText="Create adaptive card along with sample template binding";
+        return labelElement;
+    }
+
+	private checkFileSize(file) {
+		const filesize = parseInt(((file.size/1024)/1024).toFixed(4)); // MB
+		return filesize < 2;
```

Comment:
Can be moved to top as ALLOWED_SIZE constant

Reply:
Addressed the review comments